ipaserver/install/ipa_idrange_fix.py

```diff
@@ -0,0 +1,1073 @@
+
+        if not ranges_overlap_check(self.id_ranges):
+            logger.error(
+                "Ranges overlap detected, cannot proceed! Please adjust \
+existing ranges manually."
+            )
+            return 1
+
+        # Checking RID bases for existing ranges
+        id_ranges_nobase = get_ranges_no_base(self.id_ranges)
+
+        if len(id_ranges_nobase) > 0:
+            logger.info(
+                "Found %s ranges without base RIDs", len(id_ranges_nobase)
+            )
+            for id_range in id_ranges_nobase:
+                logger.debug(
+                    "Range '%s' has RID base %s and secondary RID base %s",
+                    id_range.name,
+                    id_range.base_rid,
+                    id_range.secondary_base_rid,
+                )
+            propose_rid_ranges(
+                self.id_ranges,
+                self.options.ridoffset,
+                self.propositions_rid
+            )
+        else:
+            logger.info(
+                "All ID ranges have base RIDs set, RID adjustments are \
+not needed."
+            )
+        return 0
+
+    def evaluate_identities(self, ids_out_of_range: list[IDentity]) -> None:
+        """Function to evaluate out of range IDs"""
+        if len(ids_out_of_range) == 0:
+            logger.info("No out of range IDs found!")
+        else:
+            logger.info(
+                "Found overall %s IDs out of already set up ID ranges.\n",
+                len(ids_out_of_range),
+            )
+            # ruling out IDs under 1000 if flag is not set
+            if not self.options.allowunder1000:
+                self.under1000, ids_out_of_range = separate_under1000(
+                    ids_out_of_range
+                )
+                if len(self.under1000) > 0:
+                    logger.info(
+                        "Found IDs under 1000, which is not recommeneded \
+(if you definitely need ranges proposed for those, use --allowunder1000):"
+                    )
+                    for identity in self.under1000:
+                        logger.info("%s", identity)
+
+            # Get initial divide of IDs into groups
+            groups = group_identities_by_threshold(
+                ids_out_of_range, self.options.rangegap
+            )
+
+            # Get outliers from too small groups and clean groups for
+            # further processing
+            self.outliers, cleangroups = separate_ranges_and_outliers(
+                groups, self.options.minrange
+            )
+
+            # Print the outliers, they have to be moved manually
+            if len(self.outliers) > 0:
+                print()
+                logger.info(
+                    "Identities that don't fit the criteria to get a new \
+range found! Current attributes:\n\
+Minimal range size: %s\n\
+Maximum gap between IDs: %s\n\
+Try adjusting --minrange, --rangegap or move the following identities into \
+already existing ranges:",
```

Comment:
Similar formatting note. It is a bit off-putting to not have this indented, even if allowed.

Reply:
same problem here, fixed with correct concatenation.